src/audio/dai-zephyr.c

```diff
@@ -0,0 +1,1408 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2016 Intel Corporation. All rights reserved.
+//
+// Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component_ext.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/ipc/msg.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/drivers/timer.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/cache.h>
+#include <sof/lib/dai.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/dai.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <device.h>
+#include <drivers/dai.h>
+
+static const struct comp_driver comp_dai;
+
+LOG_MODULE_REGISTER(dai_comp, CONFIG_SOF_LOG_LEVEL);
+
+/* c2b00d27-ffbc-4150-a51a-245c79c5e54b */
+DECLARE_SOF_RT_UUID("dai", dai_comp_uuid, 0xc2b00d27, 0xffbc, 0x4150,
+		 0xa5, 0x1a, 0x24, 0x5c, 0x79, 0xc5, 0xe5, 0x4b);
+
+DECLARE_TR_CTX(dai_comp_tr, SOF_UUID(dai_comp_uuid), LOG_LEVEL_INFO);
+
+#if CONFIG_INTEL_SSP
+static struct dai ssp_shared[6];
+#endif
+#if CONFIG_INTEL_ALH
+static struct dai alh_shared[2];
+#endif
+
+const struct dai_type_info dti[] = {
+#if CONFIG_INTEL_SSP
+	{
+		.type = SOF_DAI_INTEL_SSP,
+		.dai_array = cache_to_uncache_init((struct dai *)ssp_shared),
+		.num_dais = ARRAY_SIZE(ssp_shared)
+	},
+#endif
+#if CONFIG_INTEL_ALH
+	{
+		.type = SOF_DAI_INTEL_ALH,
+		.dai_array = cache_to_uncache_init((struct dai *)alh_shared),
+		.num_dais = ARRAY_SIZE(alh_shared)
+	}
+#endif
+};
+
+static int dai_trigger_op(struct dai *dai, int cmd, int direction)
```

Comment:
All the Zephyr DAI APIs start with dai_ prefix can remove this for all our static calls. i.e. this func becomes trigger_op()

Reply:
oh yes, forgot this when removed probe_op etc.

Reply:
oh but this is like trigger_internal, so the actual dai_trigger is called. So this is the actual translation from sof to zephyr

Reply:
trigger_internal is fine, we should not use dai_ prefix.

Reply:
ok